File: mobile/lib/models/field.dart
class Field {
  final String id;
  final String farmId;
  final String fieldName;
  final double areaSize;
  final String soilType;
  final String? boundaryCoordinates;

  Field({
    required this.id,
    required this.farmId,
    required this.fieldName,
    required this.areaSize,
    required this.soilType,
    this.boundaryCoordinates,
  });

  factory Field.fromJson(Map<String, dynamic> json) {
    return Field(
      id: json['id'],
      farmId: json['farmId'],
      fieldName: json['fieldName'],
      areaSize: (json['areaSize'] as num).toDouble(),
      soilType: json['soilType'],
      boundaryCoordinates: json['boundaryCoordinates'],
    );
  }

  Map<String, dynamic> toCreateJson() {
    return {
      'farmId': farmId,
      'fieldName': fieldName,
      'areaSize': areaSize,
      'soilType': soilType,
      'boundaryCoordinates': boundaryCoordinates,
    };
  }
}

File: backend/AgriOpsAI.Api/Models/Field.cs
namespace AgriOpsAI.Api.Models;

public class Field
{
    public Guid Id { get; set;}

    public Guid FarmId{ get; set;}
    public string FieldName{get; set;} = String.Empty;
    public decimal AreaSize {get; set;} 
    public string SoilType {get; set;} = String.Empty;
    public string? BoundaryCoordinates {get; set;}
    public DateTime CreatedAt {get;set;} = DateTime.UtcNow;
    public DateTime UpdatedAt {get;set;} = DateTime.UtcNow;

    public Farm? Farm {get; set;} = null!;
    public ICollection<CropSeason> CropSeasons {get;set;} = new List<CropSeason>();
    public ICollection<SoilRecord> SoilRecords {get;set;} = new List<SoilRecord>();
}